### DISCUSSION TITLE
What are the use cases to choose MariaDB/SQLite over Redis

### DISCUSSION THREAD

**Discussion by myc724** [juicedata/juicefs#718] [Q&A] [answered]
We’re using Redis as the metadata engine for a while. Since JuiceFS has extended the metadata service support in the past few months, we start to explore TiKV. Our plan is to use TiKV (over Redis) for large-scale runs on the cloud. We just wonder what are the preferred use cases for MariaDB and/or SQLite. Thanks!

**Comment by xiaogaozi** [comment, marked as answer]
If you use SQLite as the metadata engine, the data cannot be shared between multiple machines.

MariaDB/MySQL and TiKV have similar performance, but MariaDB is not as scalable as TiKV.

Please refer to ["Metadata Engines Benchmark"](https://github.com/juicedata/juicefs/blob/main/docs/en/metadata_engines_benchmark.md) for more information.

**Reply by myc724** [reply]
Got it. Thanks for the pointers.
